src/library_fs_shared.js

```diff
@@ -48,14 +48,9 @@ mergeInto(LibraryManager.library, {
 #endif
   ],
   $FS_createPreloadedFile: function(parent, name, url, canRead, canWrite, onload, onerror, dontCreateFile, canOwn, preFinish) {
-#if WASMFS
-    // TODO: use WasmFS code to resolve and join the path here?
-    var fullname = name ? parent + '/' + name : parent;
-#else
     // TODO we should allow people to just pass in a complete filename instead
     // of parent and name being that we just join them anyways
     var fullname = name ? PATH_FS.resolve(PATH.join2(parent, name)) : parent;
```

Comment:
Do we have an implementation of `PATH_FS.resolve` in WasmFS now, then? What does it do?

Reply:
It's just the normal `PATH_FS` code. It's a little odd to have that code in JS given WasmFS is in wasm, but it works.

Reply:
I think we had this workaround in there to reduce code size, since as far as we could tell, `PATH_FS.resolve` was doing a bunch of work that was unnecessary in WasmFS because WasmFS handles paths more correctly than the old FS. It sounds like this was incorrect, if things were broken before this PR. What exactly was going wrong?

Reply:
It wasn't just about code size, but that `PATH_FS` calls `FS.cwd()`, which in WasmFS needs to call into wasm. That was tricky, but it was resolved in #19452 and improved in #19616 After that, `FS.cwd()` works ok and we can resolve paths using it.

We may want to improve code size here, separately.  This code does two things: canonicalize paths so we find them in lookup maps like `preloadedFiles` (e.g., remove `/../`), and use the current working dir for `./file`. We still need to call into wasm for the latter, but we could have shorter code for the former, perhaps.

Reply:
I see, thanks!